ikarus/controlroutines/controlstate.hh

```diff
@@ -0,0 +1,42 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+/**
+ * \file controlstate.hh
+ * \brief Defines the ControlState structure for storing results of a control routine.
+ * \ingroup  controlroutines
+ */
+
+#pragma once
+
+#include <ikarus/solver/nonlinearsolver/solverstate.hh>
+
+namespace Ikarus {
+
+/**
+ * \struct ControlState
+ * \brief Structure containing information about the state of the control routine.
+ *
+ * \details This structure holds information about the success of a control routine,
+ * the flag indicating if the state of deformation is in the initial configuration,
+ * including current load step number, total number of iterations performed by the non-linear solver,
+ * load step size, value of the load factor, the solution vector,
+ * state of the non-linear solver and the name of the control routine.
+ *
+ * \ingroup  controlroutines
+ */
+struct ControlState
+{
+  bool success{false};      ///< Flag indicating the success of the control.
+  bool initialConfig{true}; ///< Flag indicating if the state of deformation is in the initial configuration
+  int currentStep{-1};      ///< Information about the current step number.
+  int totalIterations{-1};  ///< Information about the total iterations performed.
```

Comment:
It is not clear what iterations are meant. Do you mean the accumulated iterations of the non-linear solver?

Reply:
If yes this information is not needed here. It can be collected from the `std::vector<NonLinearSolverState>`

Reply:
Yes the accumulated iterations are meant here. This was existing in some previous [state](https://github.com/ikarus-project/ikarus/blob/bc987df2c0f29c29f8862459ecf0a72176c68dd5/ikarus/controlroutines/controlinfos.hh#L25), but you are right, it can be removed in this PR